src/events-extractors/ical.ts

```diff
@@ -34,3 +34,11 @@ export function extractEventsFromICal(
 		})
 		.filter((event) => event) as DayjsPeriod[]
 }
+
+function _parseICalDate(dateString: string, timeZone: string): Dayjs {
+	if (dateString.length > 15) {
```

Comment:
Maybe add a regex to prevent miscalculation on malformed data

Reply:
Malformed data will be rejected by `dayjs` and ignored by the extractor. Miscalculation should not happen without crash `dayjs` parsing and regex may slow the parsing of large calendar. What do you think @Juskie ?

Reply:
If dayjs throws, looks good to me.